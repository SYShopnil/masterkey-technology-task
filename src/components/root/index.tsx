export { SIconStore } from "./_icon/";
export { SIconWithMessage } from "./s-icon-with-message";
export { SLoading } from "./s-loading";
