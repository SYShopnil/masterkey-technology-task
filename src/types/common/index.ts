export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  SHeaderMain = "SHeaderMain",
  SProfile = "SProfile",
  CAddNewText = "CAddNewText",
  CInputCheckBoxContainer = "CInputCheckBoxContainer",
  SShowAllTextBox = "SShowAllTextBox",
  STextBox = "STextBox",
  CButton = "CButton",
  CLogout = "CLogout",
  CModal = "CModal",
  SIconWithMessage = "SIconWithMessage",
  SLoading = "SLoading",
  STextCardSkelton = "STextCardSkelton",
  ProfilePage = "ProfilePage",
}
