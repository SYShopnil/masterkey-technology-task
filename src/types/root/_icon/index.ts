export enum IconName {
  DownArrow = "DownArrow",
  GraduationCap = "GraduationCap",
  QuestionSign = "QuestionSign",
  AiOutlineSound = "AiOutlineSound",
  ArrowRight = "ArrowRight",
  DeleteSign = "TiDeleteOutline",
  Logo = "Logo",
  SiPrivateinternetaccess = "SiPrivateinternetaccess",
  TbError404 = "TbError404",
  FaArrowAltCircleRight = "FaArrowAltCircleRight",
}

export interface IIconStore {
  iconName: IconName;
  fill?: string;
}
