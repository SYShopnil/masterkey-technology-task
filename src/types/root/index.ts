export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
