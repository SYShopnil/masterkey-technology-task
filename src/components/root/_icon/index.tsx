import React from "react";

import { IoIosArrowDown } from "react-icons/io";
import { IIconStore, IconEnums } from "@src/types/root";
import { FaArrowAltCircleRight } from "react-icons/fa";
import { TbError404 } from "react-icons/tb";
import { Logo } from "./logo";

export const SIconStore = ({ iconName, fill }: IIconStore) => {
  switch (iconName) {
    case IconEnums.Logo: {
      return <Logo />;
    }
    case IconEnums.TbError404: {
      return <TbError404 fill={fill} />;
    }
    case IconEnums.FaArrowAltCircleRight: {
      return <FaArrowAltCircleRight fill={fill} />;
    }
    default: {
      return <IoIosArrowDown />;
    }
  }
};
