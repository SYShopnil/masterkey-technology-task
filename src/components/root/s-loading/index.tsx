import { EDataTestId } from "@src/types/common";

interface ISLoading {
  text: string;
}

export function SLoading({ text }: ISLoading) {
  return (
    <div
      className="flex justify-center items-center h-[50vh]"
      role={EDataTestId.SLoading}
    >
      <p className={`text-center font-extrabold text-2xl text-[#7F4D4F]`}>
        {text ? <>{text}</> : <>Loading Please Wait....</>}
      </p>
    </div>
  );
}
