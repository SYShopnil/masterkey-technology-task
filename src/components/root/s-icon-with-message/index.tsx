import React from "react";
import { SIconStore } from "../_icon";
import { IRIconWithMessage } from "@src/types/root/s-icon-with-message";
import Link from "next/link";
import { EDataTestId } from "@src/types/common";

export const SIconWithMessage = ({ icon, message }: IRIconWithMessage) => {
  return (
    <div
      className={`h-[60vh] flex justify-center items-center space-x-2`}
      role={EDataTestId.SIconWithMessage}
    >
      <div>
        <SIconStore iconName={icon} fill={"#7F4D4F"} />
      </div>
      <div>
        <p className={`font-bold`}>{message}</p>
      </div>
      <div className="!ml-8 ">
        <a className="text-[#7F4D4F] hover:text-black duration-[0.2s] font-bold">
          <Link href="/">Click Me</Link>
        </a>
      </div>
    </div>
  );
};
