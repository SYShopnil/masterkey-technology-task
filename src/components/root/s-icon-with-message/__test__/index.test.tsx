import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { SIconWithMessage } from "..";
import { IRIconWithMessage } from "@src/types/root/s-icon-with-message";
import { IconEnums } from "@src/types/root";

const mock_data: IRIconWithMessage = {
  icon: IconEnums.TbError404,
  message: "Somethings Error",
  url: "http://www.google.com",
};

describe("When icon with message component has rendered", () => {
  it("Expect render successfully", () => {
    render(<SIconWithMessage {...mock_data} />);
    const myElement = screen.getByRole(EDataTestId.SIconWithMessage);
    expect(myElement).toMatchSnapshot();
  });
});
