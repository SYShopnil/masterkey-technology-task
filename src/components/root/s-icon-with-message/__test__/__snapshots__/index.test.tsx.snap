// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When icon with message component has rendered Expect render successfully 1`] = `
<div
  class="h-[60vh] flex justify-center items-center space-x-2"
  role="SIconWithMessage"
>
  <div>
    <svg
      fill="#7F4D4F"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M3 7v4a1 1 0 0 0 1 1h3"
      />
      <path
        d="M7 7v10"
      />
      <path
        d="M10 8v8a1 1 0 0 0 1 1h2a1 1 0 0 0 1 -1v-8a1 1 0 0 0 -1 -1h-2a1 1 0 0 0 -1 1z"
      />
      <path
        d="M17 7v4a1 1 0 0 0 1 1h3"
      />
      <path
        d="M21 7v10"
      />
    </svg>
  </div>
  <div>
    <p
      class="font-bold"
    >
      Somethings Error
    </p>
  </div>
  <div
    class="!ml-8 "
  >
    <a
      class="text-[#7F4D4F] hover:text-black duration-[0.2s] font-bold"
    >
      <a
        href="/"
      >
        Click Me
      </a>
    </a>
  </div>
</div>
`;
